create_default_cpacket_dashboard.py

```diff
@@ -0,0 +1,456 @@
+        "name": str(var_name),
+        "options": [],
+        "query": str(query),
+        "refresh": 1,
+        "regex": "",
+        "skipUrlSync": False,
+        "sort": 0,
+        "tagValuesQuery": "",
+        "tags": [],
+        "tagsQuery": "",
+        "type": "query",
+        "useTags": False,
+    }
+    return var
+
+
+def network_monitor_var():
+    label = "Network Monitors to Display"
+    var_name = "Network Monitors to Display"
+    query = "show tag values with key = \"network_monitor_name\""
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def vlan_var():
+    label = "vlan"
+    var_name = "vlan"
+    query = "show tag values with key=~/vlan_.*/"
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def cvu_var():
+    label = "cvu"
+    var_name = "cvu"
+    query = "show tag values with key=\"device\""
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def port_var():
+    label = "cvu port name"
+    var_name = "port"
+    query = "show tag values with key=\"p_nm\""
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def cburst_group_name_var():
+    label = "cburst group name"
+    var_name = "group"
+    query = "show tag values with key=\"cb_group_name\""
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def ip_var():
+    label = "ip"
+    var_name = "ip"
+    query = "show tag values with key=~/.*_ip/ LIMIT 100"
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def application_var():
+    label = "application port"
+    var_name = "application"
+    query = f"show tag values with key= {APPLICATION_PORT_STR}"
+    return simple_query_variable(INDICATORS_DATASOURCE, label, var_name, query)
+
+
+def single_vlan_var(vlan_name, vlan_tag_id):
+    # this is an alternative way to generate named vlans to vlan_pairs_var
+    var = {
+        "current": {"selected": False, "text": str(vlan_tag_id), "value": str(vlan_tag_id)},
+        "error": None,
+        "hide": 2,
+        "label": None,
+        "name": str(vlan_name),
+        "options": [{"selected": True, "text": str(vlan_tag_id), "value": str(vlan_tag_id)}],
+        "query": str(vlan_tag_id),
+        "skipUrlSync": False,
+        "type": "constant",
+    }
+    return var
+
+
+def vlan_names_var(vlan_keys, var_name="vlan_names", label="VLAN Names"):
+    # this is an alternative way to generate named vlans to vlan_pairs_var
+    var = {
+        "allValue": None,
+        "current": {"selected": True, "tags": [], "text": [], "value": []},
+        "error": None,
+        "hide": 0,
+        "includeAll": True,
+        "label": str(label),
+        "multi": True,
+        "name": str(var_name),
+        "options": [],
+        "query": "",
+        "queryValue": "",
+        "skipUrlSync": False,
+        "type": "custom",
+    }
+
+    first = True
```

Comment:
When I first started writing InfluxDB queries in Python, I used a similar pattern that treated the first item differently from the rest:

select = f"SELECT {list[0]}"
for name in list[1:]:
    select += f", {name}"

I've since replaced the above approaches with the following (when the items in the list are strings):
", ".join(list)

or when the items in the list are not strings:
", ".join([f"{name}" for name in list]) 

A similar idea can be applied for comma separated key: value pairs:
", ".join([f"{key}: {value}" for key, value in dict.items()])

Given there are other operations happening inside the iteration, it probably does not make sense to attempt to apply the ", ".join() approach here.  I just thought I'd mention it in the off chance it might come in handy in the future.

Reply:
right